src/Interpreters/OptimizeDateFilterVisitor.cpp

```diff
@@ -0,0 +1,122 @@
+#include <Interpreters/OptimizeDateFilterVisitor.h>
+
+#include <Common/DateLUT.h>
+#include <Common/DateLUTImpl.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+
+
+namespace DB
+{
+
+ASTPtr generateOptimizedDateFilterAST(const String & comparator, const String & converter, const String & column, UInt64 compare_to)
+{
+    const DateLUTImpl & date_lut = DateLUT::instance();
+
+    if (converter != "toYear") return {};
+
+    UInt64 year = compare_to;
```

Comment:
Seems we don't need `compare_to` in the following code. Why not name the param `year`?

Reply:
As is shown in this [prototype](https://github.com/ZhiguoZh/ClickHouse/commit/c7f1753f0c9363a19d95fa46f1cfed1d9f505ee0), `compare_to` is declared for a later optimization of `toYYYYMM`, where `compare_to` will be a constant, yyyymm, for the comparison with `toYYYYMM`. 

But I agree with you that naming `compare_to` as `year` makes much more sense in this PR. Revised accordingly, thanks for your suggestion!